sdk/python/feast/infra/aws.py

```diff
@@ -244,52 +255,72 @@ def _upload_docker_image(self, project: str) -> str:
+        for line in docker_client.api.pull(
+            AWS_LAMBDA_FEATURE_SERVER_IMAGE, stream=True, decode=True
+        ):
+            _logger.debug(f"  {line}")
 
         auth_token = ecr_client.get_authorization_token()["authorizationData"][0][
             "authorizationToken"
         ]
         username, password = base64.b64decode(auth_token).decode("utf-8").split(":")
 
         ecr_address = repository_uri.split("/")[0]
-        docker_client.login(username=username, password=password, registry=ecr_address)
+        _logger.info(
+            f"Logging in Docker client to {Style.BRIGHT + Fore.GREEN}{ecr_address}{Style.RESET_ALL}"
+        )
+        login_status = docker_client.login(
+            username=username, password=password, registry=ecr_address
+        )
+        _logger.debug(f"  {login_status}")
 
         image = docker_client.images.get(AWS_LAMBDA_FEATURE_SERVER_IMAGE)
         image_remote_name = f"{repository_uri}:{version}"
         _logger.info(
-            f"Pushing local image to remote {Style.BRIGHT + Fore.GREEN}{image_remote_name}{Style.RESET_ALL}:"
+            f"Pushing local image to remote {Style.BRIGHT + Fore.GREEN}{image_remote_name}{Style.RESET_ALL}"
         )
         image.tag(image_remote_name)
-        docker_client.api.push(repository_uri, tag=version)
+        for line in docker_client.api.push(
+            repository_uri, tag=version, stream=True, decode=True
+        ):
+            _logger.debug(f"  {line}")
+
         return image_remote_name
 
-    def _get_lambda_name(self, project: str):
-        return f"feast-python-server-{project}-{self._get_version_for_aws()}"
+    def _create_or_get_repository_uri(self, ecr_client):
+        try:
+            return ecr_client.describe_repositories(
+                repositoryNames=[AWS_LAMBDA_FEATURE_SERVER_REPOSITORY]
+            )["repositories"][0]["repositoryUri"]
+        except ecr_client.exceptions.RepositoryNotFoundException:
+            _logger.info(
+                f"Creating remote ECR repository {Style.BRIGHT + Fore.GREEN}{AWS_LAMBDA_FEATURE_SERVER_REPOSITORY}{Style.RESET_ALL}"
+            )
+            response = ecr_client.create_repository(
+                repositoryName=AWS_LAMBDA_FEATURE_SERVER_REPOSITORY
+            )
+            return response["repository"]["repositoryUri"]
+
 
-    @staticmethod
-    def _get_version_for_aws():
-        """Returns Feast version with certain characters replaced.
+def _get_lambda_name(project: str):
+    lambda_prefix = AWS_LAMBDA_FEATURE_SERVER_REPOSITORY
+    lambda_suffix = f"{project}-{_get_version_for_aws()}"
+    # AWS Lambda name can't have the length greater than 64 bytes.
+    # This usually occurs during integration tests or when feast is
+    # installed in editable mode (pip install -e), where feast version is long
+    if len(lambda_prefix) + len(lambda_suffix) >= 63:
+        lambda_suffix = base64.b64encode(lambda_suffix.encode()).decode()[:40]
```

Comment:
is 40 enough here? looks like the prefix is 36 characters?

Reply:
isn't the prefix feast-python-server which is 19 characters? so 19+40 = 59 < 63.

Reply:
oops it is